packages/kbn-babel-plugin-apm-idx/index.js

```diff
@@ -0,0 +1,306 @@
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ */
+
+/* eslint strict: 0, new-cap: 0 */
+
+'use strict';
+module.exports = context => {
+  const t = context.types;
+
+  const idxRe = /\bidx\b/;
+
+  function checkIdxArguments(file, node) {
+    const args = node.arguments;
+    if (args.length !== 2) {
+      throw file.buildCodeFrameError(node, 'The `idx` function takes exactly two arguments.');
+    }
+    const arrowFunction = args[1];
+    if (!t.isArrowFunctionExpression(arrowFunction)) {
+      throw file.buildCodeFrameError(
+        arrowFunction,
+        'The second argument supplied to `idx` must be an arrow function.'
+      );
+    }
+    if (!t.isExpression(arrowFunction.body)) {
+      throw file.buildCodeFrameError(
+        arrowFunction.body,
+        'The body of the arrow function supplied to `idx` must be a single ' +
+          'expression (without curly braces).'
+      );
+    }
+    if (arrowFunction.params.length !== 1) {
+      throw file.buildCodeFrameError(
+        arrowFunction.params[2] || arrowFunction,
+        'The arrow function supplied to `idx` must take exactly one parameter.'
+      );
+    }
+    const input = arrowFunction.params[0];
+    if (!t.isIdentifier(input)) {
+      throw file.buildCodeFrameError(
+        arrowFunction.params[0],
+        'The parameter supplied to `idx` must be an identifier.'
+      );
+    }
+  }
+
+  function checkIdxBindingNode(file, node) {
+    if (t.isImportDeclaration(node)) {
+      // E.g. `import '...'`
+      if (node.specifiers.length === 0) {
+        throw file.buildCodeFrameError(node, 'The idx import must have a value.');
+      }
+      // E.g. `import A, {B} from '...'`
+      //      `import A, * as B from '...'`
+      //      `import {A, B} from '...'`
+      if (node.specifiers.length > 1) {
+        throw file.buildCodeFrameError(
+          node.specifiers[1],
+          'The idx import must be a single specifier.'
+        );
+      }
+      // `importKind` is not a property unless flow syntax is enabled.
+      // On specifiers, `importKind` is not "value" when it's not a type, it's
+      // `null`.
+      // E.g. `import type {...} from '...'`
+      //      `import typeof {...} from '...'`
+      //      `import {type ...} from '...'`.
+      //      `import {typeof ...} from '...'`
+      if (
+        node.importKind === 'type' ||
+        node.importKind === 'typeof' ||
+        node.specifiers[0].importKind === 'type' ||
+        node.specifiers[0].importKind === 'typeof'
+      ) {
+        throw file.buildCodeFrameError(node, 'The idx import must be a value import.');
+      }
+    } else if (t.isVariableDeclarator(node)) {
+      // E.g. var {idx} or var [idx]
+      if (!t.isIdentifier(node.id)) {
+        throw file.buildCodeFrameError(
+          node.specifiers[0],
+          'The idx declaration must be an identifier.'
+        );
+      }
+    }
+  }
+
```

Comment:
New code in lines 118 - 215 significantly changes the output of the transpiled code from the original.